lib/doorkeeper/oauth/token.rb

```diff
@@ -55,7 +59,11 @@ def self.from_request(request, *methods)
 
       def self.authenticate(request, *methods)
         if token = from_request(request, *methods)
-          AccessToken.by_token(token)
+          access_token = AccessToken.by_token(token)
+          if access_token && refresh_token_revoked_on_use?
+            access_token.revoke_previous_refresh_token!
+          end
```

Comment:
What do you think of always calling `access_token.revoke_previous_refresh_token!` here, and adding the configuration conditional in the model?